goal_src/jak2/engine/physics/trajectory.gc

```diff
@@ -7,3 +7,538 @@
+                        (the-as collide-shape-prim s2-0)
+                        )
+                    )
+             )
+        (set! (-> s3-0 cshape1) #f)
+        (set! (-> s3-0 cshape2) (if v1-12
+                                    (-> v1-12 cshape)
+                                    )
+              )
+        (set! (-> s3-0 resolve-u) (the int f30-0))
+        (set! (-> s3-0 head) #f)
+        (send-event
+          arg1
+          'impact-control
+          :from (if v1-12
+              (-> v1-12 cshape process)
+              arg1
+              )
+          #f
+          s3-0
+          obj
+          arg0
+          )
+        )
+      )
+    f30-0
+    )
+  )
+
+(defmethod initialize point-tracker ((obj point-tracker) (arg0 vector) (arg1 vector))
+  (set! (-> obj trans 0 quad) (-> arg0 quad))
+  (set! (-> obj trans 1 quad) (-> arg1 quad))
+  obj
+  )
+
+(defmethod point-tracker-method-10 point-tracker ((obj point-tracker) (arg0 vector) (arg1 vector) (arg2 vector) (arg3 float))
+  (rlet ((acc :class vf)
+         (vf0 :class vf)
+         (vf1 :class vf)
+         (vf2 :class vf)
+         (vf3 :class vf)
+         (vf4 :class vf)
+         )
+    (init-vf0-vector)
+    (cond
+      ((>= 0.0 arg3)
+       (set! (-> arg0 quad) (-> arg1 quad))
+       )
+      ((>= arg3 1.0)
+       (set! (-> arg0 quad) (-> arg2 quad))
+       )
+      (else
+        (let ((a1-1 arg0))
+          (let ((v1-3 arg2)
+                (a0-6 (vector-! (new 'stack-no-clear 'vector) arg1 arg2))
+                (f0-3 (- 1.0 arg3))
+                )
+            (.lvf vf2 (&-> a0-6 quad))
+            (.lvf vf1 (&-> v1-3 quad))
+            (let ((v1-4 f0-3))
+              (.mov vf3 v1-4)
+              )
+            )
+          (.add.x.vf vf4 vf0 vf0 :mask #b1000)
+          (.mul.x.vf acc vf2 vf3)
+          (.add.mul.w.vf vf4 vf1 vf0 acc :mask #b111)
+          (.svf (&-> a1-1 quad) vf4)
+          (vector-lerp! arg0 a1-1 arg2 arg3)
+          )
+        )
+      )
+    arg0
+    )
+  )
+
+(defmethod point-tracker-method-11 point-tracker ((obj point-tracker) (arg0 vector) (arg1 vector) (arg2 vector) (arg3 float))
+  (with-pp
+    (let ((v1-1 (point-tracker-method-10 obj (new 'stack-no-clear 'vector) arg1 arg2 arg3)))
+      (vector-! arg0 v1-1 arg1)
+      )
+    (vector-float*! arg0 arg0 (-> pp clock frames-per-second))
+    arg0
+    )
+  )
+
+;; WARN: Return type mismatch basic vs none.
+(defmethod combo-tracker-method-13 combo-tracker ((obj combo-tracker) (arg0 handle) (arg1 vector) (arg2 float) (arg3 vector) (arg4 float))
+  (cond
+    ((send-event (handle->process arg0) 'combo)
+     (let ((gp-1 (handle->process arg0)))
+       (if (type? gp-1 process-focusable)
+           (empty)
+           )
+       )
+     )
+    (else
+      (let ((s1-0 (new 'stack-no-clear 'sphere))
+            (s2-1 ((method-of-type array new)
+                   (the-as symbol (new 'stack-no-clear 'array 'collide-shape 32))
```

Comment:
I remember this working better in jak 1 but I could be wrong... I'll check more later.